llvm/lib/CodeGen/MachineInstr.cpp

```diff
@@ -576,6 +576,11 @@ uint32_t MachineInstr::copyFlagsFromInstruction(const Instruction &I) {
       MIFlags |= MachineInstr::MIFlag::NoSWrap;
     if (TI->hasNoUnsignedWrap())
       MIFlags |= MachineInstr::MIFlag::NoUWrap;
+  } else if (const GetElementPtrInst *GEP = dyn_cast<GetElementPtrInst>(&I)) {
+    if (GEP->hasNoUnsignedSignedWrap())
+      MIFlags |= MachineInstr::MIFlag::NoSWrap;
```

Comment:
Is it equivalent to `nusw`?

Reply:
See 
https://github.com/llvm/llvm-project/blob/73e07e924470ebab76a634e41fadf425a859e0ea/llvm/include/llvm/IR/Instructions.h#L1186

Reply:
The comment reads:
/// Determine whether the GEP has the nusw flag.

Reply:
That's right. But you're setting an nsw flag. Shouldn't we introduce nusw?

Reply:
The comment reads:
/// Determine whether the GEP has the nusw flag.

Reply:
Yes, nusw and nsw are not the same thing. You *can* transfer nsw to the scale multiplication, but not to the PTRADD, as this PR appears to be doing.

Reply:
https://llvm.org/doxygen/classllvm_1_1MachineInstr.html
uses abbrevations.